linkerd/main/src/main/scala/io/buoyant/Linkerd.scala

```diff
@@ -73,7 +52,77 @@ object Linkerd extends App {
         if (!f.isFile) throw new IllegalArgumentException(s"config is not a file: $path")
         Source.fromFile(f).mkString
     }
-
     Linker.parse(configText)
   }
+
+  private def initAdmin(linker: Linker, linkerConfig: LinkerConfig): Closable with Awaitable[Unit] = {
+    val linkerdAdmin = new LinkerdAdmin(this, linker, linkerConfig)
+    val adminInitializer = new AdminInitializer(linker.admin, linkerdAdmin.adminMuxer)
+    adminInitializer.startServer()
+    adminInitializer.adminHttpServer
+  }
+
+  private def initRouter(config: Router): Closable with Awaitable[Unit] = {
+    val router = config.initialize()
+
+    val servers = router.servers.map { server =>
+      log.info("serving %s on %s:%d", server.router, server.ip, server.port)
+      val listening = server.serve()
+      val announcements = announce(router.announcers, server)
+      Closable.sequence(announcements, listening)
+    }
+
+    new Closable with CloseAwaitably {
+      private[this] val closer = Closable.sequence(Closable.all(servers: _*), router)
+      def close(deadline: Time) = closeAwaitably { closer.close(deadline) }
+    }
+  }
+
+  private def announce(announcers: Seq[(Path, Announcer)], server: Server.Initializer): Closable =
+    Closable.all(server.announce.map(announce(announcers, server, _)): _*)
+
+  private def announce(
+    announcers: Seq[(Path, Announcer)],
+    server: Server.Initializer,
+    name: Path
+  ): Closable =
+    announcers.filter { case (pfx, _) => name.startsWith(pfx) } match {
+      case Seq() =>
+        log.warning("no announcer found for %s", name.show)
+        Closable.nop
+
+      case announcers =>
```

Comment:
TIOLI: I know shadowing is a fairly common practice, but I think it's especially dangerous when the shadowed variables have the same type.  Here `announcers` refers to both the filtered and unfiltered seq of announcers, depending on the scope.